src/diff/index.js

```diff
@@ -140,9 +140,25 @@ export function diff(
 						commitQueue.push(c);
 					}
 
+					let lastVNodeChild;
 					for (tmp = 0; tmp < newVNode._children.length; tmp++) {
 						if (newVNode._children[tmp]) {
 							newVNode._children[tmp]._parent = newVNode;
+							lastVNodeChild = newVNode._children[tmp];
+						}
+					}
+
+					if (
+						lastVNodeChild &&
+						lastVNodeChild.type &&
+						lastVNodeChild._dom &&
+						lastVNodeChild._dom.isConnected
+					) {
+						newVNode._nextDom = getDomSibling(lastVNodeChild);
+						if (newVNode._nextDom !== newVNode._dom) {
+							newVNode._children.forEach(() => {
+								// Reorder if needed
```

Comment:
As researched by @andrewiggins wee should traverse the children here and apply the logic for diffChildren to assess if we're ordering correctly